frontend/src/components/widgets/NumberInput/NumberInput.tsx

```diff
@@ -245,11 +262,18 @@ class NumberInput extends React.PureComponent<Props, State> {
   }
 
   public render = (): React.ReactNode => {
-    const { element, width, disabled } = this.props
+    const { element, width, disabled, widgetMgr } = this.props
     const { formattedValue, dirty } = this.state
 
     const style = { width }
 
+    // Manage our form-clear event handler.
+    this.formClearHelper.useFormClearListener(
```

Comment:
- Why should this be called inside `render`?
- Can we move this logic to `componentDidMount` and `componentDidUpdate`?

Reply:
This just follows the hooks pattern - it's logic that should be call on every update, and `render` is the function that's also called on every update. It _could_ be moved into `componentDidMount` and `componentDidUpdate` - but this wouldn't change the functionality, and would introduce another call site.